### DISCUSSION TITLE
vscode problem

### DISCUSSION THREAD

**Discussion by Soheilprs** [Cyfrin/foundry-full-course-cu#416] [Q&A]
why my vscode is like this?
![Screenshot from 2023-07-23 09-48-36](https://github.com/Cyfrin/foundry-full-course-f23/assets/106927616/4968c00c-c123-4e45-90d1-3a63abea7f4d)

**Comment by ghost** [comment]
Did you download the required extensions?

**Reply by Soheilprs** [reply]
i think so, what is the requrie extensions?

**Reply by john-lawniczak** [reply]
Type `Solidity` in the sidebar on the far left. I have this one installed:

<img width="430" alt="Screenshot 2023-07-23 at 11 29 14 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/105305546/a956292e-e210-4f71-ad51-21cfbeb4284e">

Hopefully that fixes your problem. Keep us posted.
